rdmo/projects/filters.py

```diff
@@ -13,6 +18,88 @@ class Meta:
         fields = ('title', 'catalog')
 
 
+class ProjectSearchFilterBackend(SearchFilter):
+
+    def filter_queryset(self, request, queryset, view):
+        if view.detail:
+            return queryset
+
+        search_terms = self.get_search_terms(request)
+        if search_terms:
+            for search_term in search_terms:
+                queryset = queryset.filter(
+                    Q(title__icontains=search_term) | (
+                        Q(memberships__role='owner') & (
+                            Q(memberships__user__username__icontains=search_term) |
+                            Q(memberships__user__first_name__icontains=search_term) |
+                            Q(memberships__user__last_name__icontains=search_term) |
+                            Q(memberships__user__email__icontains=search_term)
+                        )
+                    )
+                )
+
+        return queryset
+
+
+class ProjectDateFilterBackend(BaseFilterBackend):
+
+    def filter_queryset(self, request, queryset, view):
+        if view.detail:
+            return queryset
+
+        before = self.parse_query_datetime(request, 'before')
+        if before:
+            queryset = queryset.filter(updated__lte=before)
+
+        after = self.parse_query_datetime(request, 'after')
+        if after:
+            queryset = queryset.filter(updated__gte=after)
+
+        return queryset
+
+    def parse_query_datetime(self, request, key):
+        value = request.GET.get(key)
+        if value:
+            datetime = parse_datetime(value)
+
+            if not is_aware(datetime):
+                datetime = make_aware(datetime)
+
+            return datetime
+
+
+class ProjectOrderingFilter(OrderingFilter):
```

Comment:
This allows for ordering like this http://localhost:8000/api/v1/projects/projects/?ordering=owner. For `owner`, `progres`, and `role` the query needs to be annotated first.